ops/model.py

```diff
@@ -996,10 +997,15 @@ def relation_get(self, relation_id, member_name, is_app):
         if not isinstance(is_app, bool):
             raise TypeError('is_app parameter to relation_get must be a boolean')
 
+        if is_app and not JujuVersion.from_environ().has_app_data():
+            raise RuntimeError('trying to relation-get app data on a Juju too old for it')
```

Comment:
what about:
'relation-get of application data is not supported by this version of Juju {}'.format(version)